### PR TITLE
cifar10_4000跑出论文精度代码

### PR DESCRIPTION
**Pull request by zh-hike**
跑出论文精度的代码版本1，目前缺少一些代码规范，有些代码还需要整理。

### CODE REVIEW

File: ppcls/arch/backbone/model_zoo/wideresnet.py [hunk 1]
```diff
@@ -201,7 +201,7 @@ def forward(self, x):
         feat = self.relu(self.bn1(feat))
         feat = F.adaptive_avg_pool2d(feat, 1)
         feat = paddle.reshape(feat, [-1, self.channels])
-
+
```
Comment (RainFrost1) [inline ppcls/arch/backbone/model_zoo/wideresnet.py]: 空白删掉

File: ppcls/arch/gears/__init__.py [hunk 2]
```diff
@@ -20,14 +20,15 @@
...
+from .decoup import Decoup
 
 __all__ = ['build_gear']
 
 
 def build_gear(config):
     support_dict = [
         'ArcMargin', 'CosMargin', 'CircleMargin', 'FC', 'VehicleNeck', 'Tanh',
-        'BNNeck', 'AdaMargin'
+        'BNNeck', 'AdaMargin', 'FRFBNeck', 'Decoup'
```
Comment (RainFrost1) [inline ppcls/arch/gears/__init__.py]: FRFBNeck没有使用到？

File: ppcls/configs/ssl/FixMatchCCSSL/FixMatchCCSSL_cifar10_4000_4gpu.yaml [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+        weight: 1.0
+        temperature: 0.07
+    # - CCSSLLoss:
+    #     CELoss:
+    #       weight: 1.0
+    #       reduction: "none"
+    #     SoftSupConLoss:
+    #       weight: 1.0
+    #       temperature: 0.07
+    #     weight: 1.
```
Comment (RainFrost1) [inline ppcls/configs/ssl/FixMatchCCSSL/FixMatchCCSSL_cifar10_4000_4gpu.yaml]: 无用的注视最好删掉，除非有特殊情况

File: ppcls/configs/ssl/FixMatchCCSSL/FixMatchCCSSL_cifar10_4000_4gpu.yaml [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+  weight_decay: 0.001
+  lr:
+    name: 'cosine_schedule_with_warmup'
+    learning_rate: 0.03
+    num_warmup_steps: 0
+    num_training_steps: 524800
+
+DataLoader:
+  mean: &sign_mean [0.4914, 0.4822, 0.4465]
+  std: &sign_std [0.2471, 0.2435, 0.2616]
```
Comment (RainFrost1) [inline ppcls/configs/ssl/FixMatchCCSSL/FixMatchCCSSL_cifar10_4000_4gpu.yaml]: 为什么这里有个mean、std？下面用的话，直接拿到下面

File: ppcls/configs/ssl/FixMatchCCSSL/FixMatchCCSSL_cifar10_4000_4gpu.yaml [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+                  contrast: 0.4
+                  saturation: 0.4
+                  hue: 0.1
+            p: 0.8
+        - RandomGrayscale:
+            p: 0.2
+        - ToTensor:
+        # - Normalize:
+        #     mean: *sign_mean
+        #     std: *sign_std
```
Comment (RainFrost1) [inline ppcls/configs/ssl/FixMatchCCSSL/FixMatchCCSSL_cifar10_4000_4gpu.yaml]: 同上，无用注释，最好删掉

File: ppcls/arch/__init__.py [hunk 6]
```diff
@@ -14,6 +14,7 @@
 
 import copy
 import importlib
+from pyexpat import features
```
Comment (RainFrost1) [inline ppcls/arch/__init__.py]: 这个引用是？

File: ppcls/arch/__init__.py [hunk 7]
```diff
@@ -71,6 +72,9 @@ def __init__(self, **config):
         super().__init__()
         backbone_config = config["Backbone"]
         backbone_name = backbone_config.pop("name")
+        self.decoup = False
```
Comment (RainFrost1) [inline ppcls/arch/__init__.py]: Recmodel的这个不是讨论，在forward函数中加个head_featrue_from参数就行了吗？

File: ppcls/arch/gears/__init__.py [hunk 8]
```diff
@@ -27,7 +27,7 @@
 def build_gear(config):
     support_dict = [
         'ArcMargin', 'CosMargin', 'CircleMargin', 'FC', 'VehicleNeck', 'Tanh',
-        'BNNeck', 'AdaMargin'
+        'BNNeck', 'AdaMargin',
```
Comment (RainFrost1) [inline ppcls/arch/gears/__init__.py]: 你自己添加的Neck呢？

File: ppcls/configs/ssl/FixMatchCCSSL/FixMatchCCSSL_cifar100_10000_4gpu.yaml [hunk 9]
```diff
@@ -0,0 +1,204 @@
...
+        weight: 1.0
+  
+UnLabelLoss:
+  Train:
+    - CCSSLCeLoss:
+        weight: 1.
+    - SoftSupConLoss:
+        weight: 1.0
+        temperature: 0.07
+    # - CCSSLLoss:
```
Comment (RainFrost1) [inline ppcls/configs/ssl/FixMatchCCSSL/FixMatchCCSSL_cifar100_10000_4gpu.yaml]: 没用的删掉

File: ppcls/configs/ssl/FixMatchCCSSL/FixMatchCCSSL_cifar100_10000_4gpu.yaml [hunk 10]
```diff
@@ -0,0 +1,204 @@
...
+    name: WideResNet
+    widen_factor: 8
+    depth: 28
+    dropout: 0     # CCSSL为 drop_rate
+    num_classes: &sign_num_classes 100
+    low_dim: 64
+    proj: true
+    proj_after: false
+
+  Decoup:
```
Comment (RainFrost1) [inline ppcls/configs/ssl/FixMatchCCSSL/FixMatchCCSSL_cifar100_10000_4gpu.yaml]: 修改成Neck、head方式

File: ppcls/configs/ssl/FixMatchCCSSL/FixMatchCCSSL_cifar100_10000_4gpu.yaml [hunk 11]
```diff
@@ -0,0 +1,204 @@
...
+  weight_decay: 0.001
+  lr:
+    name: 'cosine_schedule_with_warmup'
+    learning_rate: 0.03
+    num_warmup_steps: 0
+    num_training_steps: 524800
+
+DataLoader:
+  mean: &sign_mean [0.5071, 0.4867, 0.4408]
+  std: &sign_std [0.2675, 0.2565, 0.2761]
```
Comment (RainFrost1) [inline ppcls/configs/ssl/FixMatchCCSSL/FixMatchCCSSL_cifar100_10000_4gpu.yaml]: 按照讨论的修改方式改

File: ppcls/configs/ssl/FixMatchCCSSL/FixMatchCCSSL_cifar100_10000_4gpu.yaml [hunk 12]
```diff
@@ -0,0 +1,204 @@
...
+                  contrast: 0.4
+                  saturation: 0.4
+                  hue: 0.1
+            p: 0.8
+        - RandomGrayscale:
+            p: 0.2
+        - ToTensor:
+        # - Normalize:
+        #     mean: *sign_mean
+        #     std: *sign_std
```
Comment (RainFrost1) [inline ppcls/configs/ssl/FixMatchCCSSL/FixMatchCCSSL_cifar100_10000_4gpu.yaml]: 没用的删掉

File: ppcls/configs/ssl/FixMatchCCSSL/FixMatchCCSSL_cifar10_4000_4gpu.yaml [hunk 13]
```diff
@@ -0,0 +1,204 @@
+Global:
```
Comment (RainFrost1) [inline ppcls/configs/ssl/FixMatchCCSSL/FixMatchCCSSL_cifar10_4000_4gpu.yaml]: 修改同上个yaml

File: ppcls/data/preprocess/__init__.py [hunk 14]
```diff
@@ -49,6 +51,8 @@
 from ppcls.data.preprocess.batch_ops.batch_operators import MixupOperator, CutmixOperator, OpSampler, FmixOperator
 from ppcls.data.preprocess.batch_ops.batch_operators import MixupCutmixHybrid
 
+
```
Comment (RainFrost1) [inline ppcls/data/preprocess/__init__.py]: 删掉多余空行

File: ppcls/data/preprocess/__init__.py [hunk 15]
```diff
@@ -117,3 +121,4 @@ def __call__(self, img):
             img = np.asarray(img)
 
         return img
+
```
Comment (RainFrost1) [inline ppcls/data/preprocess/__init__.py]: 删掉多余空行

File: ppcls/data/preprocess/ops/randaugmentmc.py [hunk 16]
```diff
@@ -0,0 +1,227 @@
+import PIL
```
Comment (RainFrost1) [inline ppcls/data/preprocess/ops/randaugmentmc.py]: 注意格式，这个跟paddleclas中的randomaugment有什么区别？

File: ppcls/engine/train/train_fixmatch_ccssl.py [hunk 17]
```diff
@@ -0,0 +1,139 @@
...
+import time
+from turtle import update
+import paddle
+from ppcls.engine.train.train_fixmatch import get_loss
+from ppcls.engine.train.utils import update_loss, update_metric, log_info
+from ppcls.utils import profiler
+from paddle.nn import functional as F
+import numpy as np
+import paddle
+# from reprod_log import ReprodLogger
```
Comment (RainFrost1) [inline ppcls/engine/train/train_fixmatch_ccssl.py]: 删掉

File: ppcls/engine/train/train_fixmatch_ccssl.py [hunk 18]
```diff
@@ -0,0 +1,139 @@
+
```
Comment (RainFrost1) [inline ppcls/engine/train/train_fixmatch_ccssl.py]: 为什么有空行？

File: ppcls/engine/train/train_fixmatch_ccssl.py [hunk 19]
```diff
@@ -0,0 +1,139 @@
...
+    for k, v in loss_dict_label.items():
+        if k != 'loss':
+            loss_dict[k] = v
+    for k, v in unlabel_loss.items():
+        if k != 'loss':
+            loss_dict[k] = v
+    loss_dict['loss'] = loss_dict_label['loss'] + unlabel_loss['loss']
+
+    return loss_dict, logits_x
+
```
Comment (RainFrost1) [inline ppcls/engine/train/train_fixmatch_ccssl.py]: 空行

File: ppcls/loss/softsuploss.py [hunk 20]
```diff
@@ -0,0 +1,69 @@
+"""
```
Comment (RainFrost1) [inline ppcls/loss/softsuploss.py]: 文件头的，写法改成其他的，看查看其他文件的写法。一般写相关的开源协议。直接拷贝过来的代码，请注明引用链接

File: ppcls/loss/softsuploss.py [hunk 21]
```diff
@@ -0,0 +1,69 @@
...
+        """
+        max_probs = batch['max_probs']
+        labels = batch['p_targets_u_w']
+        # reduction = batch['reduction']
+        batch_size = feat.shape[0]
+        if labels is not None:
+            labels = labels.reshape((-1, 1))
+            mask = paddle.equal(labels, labels.T).astype('float32')
+            max_probs = max_probs.reshape((-1, 1))
+
```
Comment (RainFrost1) [inline ppcls/loss/softsuploss.py]: 没必要的空格都删一下

File: ppcls/arch/__init__.py (hunk 7 above)
Comment (zh-hike) [reply ppcls/arch/__init__.py]: 这个我写的时候想了一下，forward函数不仅仅在train的时候调用，在验证的时候也要调用，我在train的时候可以控制head_featrue_from参数，但是验证的代码是固定的这个我没办改变

File: ppcls/arch/gears/__init__.py (hunk 8 above)
Comment (zh-hike) [reply ppcls/arch/gears/__init__.py]: 修改过RecModel后我自己就不需要Neck了

File: ppcls/data/preprocess/ops/randaugmentmc.py (hunk 16 above)
Comment (zh-hike) [reply ppcls/data/preprocess/ops/randaugmentmc.py]: 这个我代码里已经用了randomaugment，然后现在测试新的数据增强有没有改变精度，下一个pr会删掉这个randaugmentmc.py

**Review by RainFrost1 — DISMISSED** [review]

**Review by RainFrost1 — APPROVED** [review]
